inc/class-base-command.php

```diff
@@ -0,0 +1,153 @@
+		 *                          - 'min-active-sites': Minimum number of active sites required for inclusion.
+		 *                          - 'max-active-sites': Maximum number of active sites allowed for inclusion.
+		 *                          - 'site-id': Filter by a specific site ID.
+		 * @param string $column_name The name of the primary column to display in the table.
+		 *
+		 * @return array A filtered and sorted array of data, where each entry contains:
+		 *               - The primary column key with the plugin/theme name.
+		 *               - 'Active Sites': The count of sites where the plugin/theme is active.
+		 *               - 'Site IDs': A comma-separated list of site IDs where the plugin/theme is active.
+		 */
+		public function get_table( array $list, array $assoc_args, string $column_name ) : array {
+			$active_counts = array_fill_keys( $list, 0 );
+			$active_sites = array_fill_keys( $list, [] );
+
+			// Count active themes per site and store site IDs
+			foreach ( $this->sites as $site ) {
+				$plugins_or_themes = $this->before_count( $site );
+
+				foreach ( $plugins_or_themes as $plugin_or_theme ) {
+					if ( isset( $active_counts[ $plugin_or_theme ] ) ) {
+						$active_counts[ $plugin_or_theme ] ++;
+					} else {
+						$active_counts[ $plugin_or_theme ] = 1;
+					}
+
+					if ( isset( $active_sites[ $plugin_or_theme ] ) ) {
+						$active_sites[ $plugin_or_theme ][] = $site->blog_id;
+					} else {
+						$active_sites[ $plugin_or_theme ] = [ $site->blog_id ];
+					}
+				}
+
+				$this->after_count();
+			}
+
+			// Apply filters
+			$order_by = $assoc_args['order-by'] ?? 'name';
+			$min_sites_active = isset( $assoc_args['min-active-sites'] ) ? (int) $assoc_args['min-active-sites'] : null;
+			$max_sites_active = isset( $assoc_args['max-active-sites'] ) ? (int) $assoc_args['max-active-sites'] : null;
+			$filter_site_id = isset( $assoc_args['site-id'] ) ? (int) $assoc_args['site-id'] : null;
+
+			$table = [];
+			foreach ( $active_counts as $plugin_or_theme => $count ) {
+				// Apply min/max filters
+				if (
+					( $min_sites_active !== null && $count < $min_sites_active ) ||
+					( $max_sites_active !== null && $count > $max_sites_active )
+				) {
+					continue;
+				}
+
+				// Apply site ID filter, if provided
+				if (
+					$filter_site_id !== null &&
+					( empty( $active_sites[ $plugin_or_theme ] ) || ! in_array( $filter_site_id, $active_sites[ $plugin_or_theme ], false ) )
+				) {
+					continue;
+				}
+
+				$table[] = [
+					$column_name => $plugin_or_theme,
+					'Active Sites' => $count,
+					'Site IDs' => empty( $active_sites[ $plugin_or_theme ] ) ? 'None' : implode( ', ', $active_sites[ $plugin_or_theme ] ),
+				];
+			}
+
+			// Sort the table
+			if ( $order_by === 'active-sites' ) {
+				usort( $table, static function ( $a, $b ) {
+					return $b['Active Sites'] <=> $a['Active Sites'];
+				} );
```

Comment:
Might be worth considering a short function here with the spaceship operator (and below)?
```suggestion
				usort( $table, fn ( $a, $b ) => ( $b['Active Sites'] <=> $a['Active Sites'] ) );
```

Reply:
refactored out a sorting function